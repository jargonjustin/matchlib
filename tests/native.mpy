
def match_native(obj):
    match obj with:
        case "testing":
            return "test string"
        case x if isinstance(x, str):
            return "string"
        case [1, 2, 3]:
            return "int list"
        case (1, x):
            return str(x)
        case { "foo": bar }:
            return "foo: %s" % bar

if __name__ == '__main__':
    print match_native("testing")
    print match_native("foobar")
    print match_native([1,2,3])
    print match_native((1, 19))
    print match_native({"foo":42})
