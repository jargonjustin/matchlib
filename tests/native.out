test string
string
int list
19
foo: 42
